### ISSUE TITLE
why model file is large and training slower than xgboost and lightgbm

### ISSUE THREAD

**Issue by yangzhen92** [catboost/catboost#960]
Problem: slow and very large model file
catboost version: 0.15.2
Operating System: windows10
CPU: i7-8750h
GPU: gtx1060

My parameters' setting as follow, and i have try many times and found out catboost is much slower than the other two model.I don't know what is the cause.

Besides, the model file size is very large:
catboost 3.09GB
lightgbm 62.1MB
xgboost 7.8MB

Compare on the final score(kaggle ieee-cis competition), catboost is the highest though, but training time cost too much. Does someone have any idea about this??

cb_fixed_params = {
    'grow_policy': 'SymmetricTree',
    'bagging_temperature': 1,  
    'score_function': 'Cosine',  
    'nan_mode': 'Min', 
    'boosting_type': 'Plain', 
    'one_hot_max_size': 2,  
    'task_type': 'GPU',
    'loss_function': 'Logloss', 
    'eval_metric': 'AUC',
    'thread_count': -1, 
    'random_seed': 666
}
| score | params | time
|0.9236 | {'l2_leaf_reg': 6.7248, 'learning_rate': 0.0351, 'max_depth': 5, 'n_estimators': 449} | 1161.3325

xgb_fixed_params = {
    'objective': 'binary:logistic',
    'eval_metric': 'auc',
    'n_jobs': -1,
    'importance_type': 'total_gain',
    'random_state': 666,
    'tree_method': 'gpu_hist',
}
| score | params | time
| 0.9078 | {'colsample_bylevel': 0.5047, 'learning_rate': 0.0515, 'max_depth': 9, 'n_estimators': 496, 'reg_alpha': 6.3249, 'reg_lambda': 5.6142, 'subsample': 0.6166} | 129.1389

lgb_fixed_params = {
    'task': 'train',
    'objective': 'binary',
    'boosting_type': 'gbdt',
    'n_jobs': -1,
    'device_type': 'cpu',
    'random_state': 666,
    'class_weight': 'balanced',
    'metric': 'auc',
    'silent': True,
    'use_missing': True,
    'importance_type': 'gain', 
    'max_depth': -1, 
    'subsample_for_bin': 1000000, 
}
| score | params | time
| 0.8965 | {'colsample_bytree': 0.8767, 'learning_rate': 0.2008, 'n_estimators': 469, 'num_leaves': 177, 'reg_alpha': 0.458, 'reg_lambda': 0.2424, 'subsample': 0.8369} | 300.9396

**Comment by wboleksii** [comment]
I believe there is a difference in growth policies - the default choice for xgboost is depthwise, so you can't directly compare it with the symmetric tree.

**Comment by annaveronika** [comment]
The main thing here is that the dataset, that you are using contains categorical features.
This is the reason for speed and model size differences.
If you have categorical features, than catboost calculates a set of new numerical features based on categorical features and their combinations. This slows down the training and gives you performance boost.

The resulting model contains information for each categorical feature value in hash tables. This leads to large models, if your categorical feature has many different values.

There is a set of things you can do to speed up the training and to reduce the model size.
Here are the instructions:
https://catboost.ai/docs/concepts/speed-up-training.html
https://catboost.ai/docs/concepts/faq.html#reduce-the-size-of-the-final-model

**Comment by yangzhen92** [comment]
> The main thing here is that the dataset, that you are using contains categorical features.
> This is the reason for speed and model size differences.
> If you have categorical features, than catboost calculates a set of new numerical features based on categorical features and their combinations. This gives you a quality boost, but slows down the training.
> 
> The resulting model contains information for each categorical feature value in hash tables. This leads to large models, if your categorical feature has many different values.
> 
> There is a set of things you can do to speed up the training and to reduce the model size.
> Here are the instructions:
> https://catboost.ai/docs/concepts/speed-up-training.html
> https://catboost.ai/docs/concepts/faq.html#reduce-the-size-of-the-final-model

Thank you, now i understand the reason behind. The dataset contains a lot categorical features, the combination of those feature do improve the score but cost more time.
